### PR TITLE
Adding Word Embeddings Model

### PR DESCRIPTION
**Pull request by sachinsharma9780**
Added a word embeddings model in the model repository of triton inference server which provides sentence embeddings given an input text. 
The model is written using python backend which performs preprocessing of text at the Triton server side instead of doing at client side.  It uses paraphrase-mpnet-base-v2 for the best quality from the sentence transformer library (https://www.sbert.net/docs/pretrained_models.html).

### CODE REVIEW

File: python_backend_models/python_backend/examples/word_embeddings/client.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                                outputs=outputs)
+
+        result = response.get_response()
+        np_response = response.as_numpy("OUTPUT0").astype(float)
+        return np_response
+        #print(np_response.shape)
+
+embeddings = []
+for sent in example_input:
+    emb = run_inference(sent)
```
Comment (geenen124) [inline python_backend_models/python_backend/examples/word_embeddings/client.py:32]: Does this currently support batched requests? (Sending multiple sentences in a single request)

File: python_backend_models/Dockerfile [hunk 2]
```diff
@@ -3,7 +3,7 @@ FROM nvcr.io/nvidia/tritonserver:21.05-py3
 COPY ./python_backend python_backend
 RUN cd python_backend \
     && /usr/bin/python3 -m pip install --upgrade pip \
-    && pip install --no-cache-dir torch==1.8.1 transformers==4.6.0 tokenizers==0.10.2 
+    && pip install --no-cache-dir torch==1.8.1 transformers==4.6.0 tokenizers==0.10.2 sentence-transformers
```
Comment (geenen124) [inline python_backend_models/Dockerfile]: Can you pin a version for `sentence-transformers` here?

File: python_backend_models/python_backend/models/word_embeddings/config.pbtxt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      name: "INPUT0"
+      data_type: TYPE_STRING
+      dims: [-1]
+    }
+]
+output [
+    {
+      name: "OUTPUT0"
+      data_type: TYPE_FP32
+      dims: [1, 768]
```
Comment (geenen124) [inline python_backend_models/python_backend/models/word_embeddings/config.pbtxt]:
```suggestion
      dims: [-1, 768]
```
To handle multiple strings? :)

File: python_backend_models/python_backend/models/word_embeddings/1/model.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        """
+
+        responses = []
+
+        # Every Python backend must iterate over everyone of the requests
+        # and create a pb_utils.InferenceResponse for each of them.
+        for request in requests:
+            # Get INPUT0
+            in_0 = pb_utils.get_input_tensor_by_name(request, "INPUT0")
+            in_0 = in_0.as_numpy()[0].decode("utf-8")
```
Comment (geenen124) [inline python_backend_models/python_backend/models/word_embeddings/1/model.py]: It looks like this is grabbing the first input sentence in a batch? Could change this to something like 
```
in_0 = [v.decode("utf-8") for v in in_0.as_numpy()]
```
to handle multiple strings in a batch?

**Review by geenen124 — COMMENTED** [review]
Looks good! Just have a couple of small questions + suggestions regarding batched request support 😄

File: python_backend_models/python_backend/examples/word_embeddings/client.py (hunk 1 above)
Comment (sachinsharma9780) [reply python_backend_models/python_backend/examples/word_embeddings/client.py:32]: I will test the batch support might need to perform some changes in config file.

File: python_backend_models/Dockerfile (hunk 2 above)
Comment (sachinsharma9780) [reply python_backend_models/Dockerfile]: Yes sure !

File: python_backend_models/python_backend/models/word_embeddings/config.pbtxt (hunk 3 above)
Comment (sachinsharma9780) [reply python_backend_models/python_backend/models/word_embeddings/config.pbtxt]: will test it :)

File: python_backend_models/python_backend/examples/word_embeddings/client.py (hunk 1 above)
Comment (sachinsharma9780) [reply python_backend_models/python_backend/examples/word_embeddings/client.py:32]: Batch support added. Thanks @geenen124 for the support.

File: python_backend_models/python_backend/models/word_embeddings/config.pbtxt (hunk 3 above)
Comment (sachinsharma9780) [reply python_backend_models/python_backend/models/word_embeddings/config.pbtxt]: yes it worked!

File: python_backend_models/python_backend/models/word_embeddings/1/model.py (hunk 4 above)
Comment (sachinsharma9780) [reply python_backend_models/python_backend/models/word_embeddings/1/model.py]: It worked after creating a 'run_model' method (doing decoding here) outside of the 'execute' method.

**Comment by geenen124** [comment]
LGTM! 👍

File: python_backend_models/Dockerfile (hunk 2 above)
Comment (sachinsharma9780) [reply python_backend_models/Dockerfile]: added sentence-transformer version.

**Pull request merged**